sim/core/buffs.go

```diff
@@ -1144,3 +1145,153 @@ func ManaTideTotemAura(character *Character, actionTag int32) *Aura {
+	if numStormLashes == 0 {
+		return
+	}
+
+	sbAura := StormLashAura(agent.GetCharacter(), -1)
+
+	registerExternalConsecutiveCDApproximation(
+		agent,
+		externalConsecutiveCDApproximation{
+			ActionID:         ActionID{SpellID: 120668, Tag: -1},
+			AuraTag:          StormLashAuraTag,
+			CooldownPriority: CooldownPriorityDefault,
+			AuraDuration:     StormLashDuration,
+			AuraCD:           StormLashCD,
+			Type:             CooldownTypeDPS,
+
+			ShouldActivate: func(sim *Simulation, character *Character) bool {
+				return true
+			},
+			AddAura: func(sim *Simulation, character *Character) {
+				sbAura.Activate(sim)
+			},
+		},
+		numStormLashes)
+}
+
+var StormLashSpellExceptions = map[int32]float64{
+	1120:   2.0, // Drain Soul
+	45284:  2.0, // Lightning Bolt
+	51505:  2.0, // Lava Burst
+	103103: 2.0, // Malefic Grasp
+	15407:  1.0, // Mind Flay
+	129197: 1.0, // Mind Flay - Insanity
+}
+
+// Source: https://www.wowhead.com/mop-classic/spell=120668/stormlash-totem#comments
+func StormLashAura(character *Character, actionTag int32) *Aura {
+	for _, pet := range character.Pets {
+		if !pet.IsGuardian() {
+			StormLashAura(&pet.Character, actionTag)
+		}
+	}
+
+	damage := 0.0
+
+	stormlashSpell := character.RegisterSpell(SpellConfig{
+		ActionID:    ActionID{SpellID: 120687, Tag: actionTag},
+		Flags:       SpellFlagNoOnCastComplete | SpellFlagPassiveSpell,
+		SpellSchool: SpellSchoolNature,
+		ProcMask:    ProcMaskEmpty,
+
+		DamageMultiplier: 1,
+		CritMultiplier:   character.DefaultCritMultiplier(),
+
+		ApplyEffects: func(sim *Simulation, target *Unit, spell *Spell) {
+			spell.CalcAndDealDamage(sim, target, damage, spell.OutcomeMagicHitAndCrit)
+		},
+	})
+
+	getStormLashSpellOverride := func(spell *Spell) float64 {
+		return StormLashSpellExceptions[spell.ActionID.SpellID]
+	}
+
+	handler := func(aura *Aura, sim *Simulation, spell *Spell, result *SpellResult) {
+		if !aura.Icd.IsReady(sim) || !result.Landed() || !spell.ProcMask.Matches(ProcMaskDirect|ProcMaskSpecial) || !sim.Proc(0.5, "Stormlash") {
+			return
+		}
+
+		baseMultiplierExtension := getStormLashSpellOverride(spell)
+		ap := Ternary(spell.ProcMask.Matches(ProcMaskRanged), stormlashSpell.RangedAttackPower(result.Target), stormlashSpell.MeleeAttackPower())
+		sp := stormlashSpell.SpellPower()
+
+		baseDamage := max(ap*0.2, sp*0.3)
+		baseMultiplier := 2.0
+		speedMultiplier := 1.0
+		if baseMultiplierExtension != 0 {
+			baseMultiplier = baseMultiplier * baseMultiplierExtension
+		}
+		if spell.Unit.Type == PetUnit {
+			baseMultiplier *= 0.2
+		}
+
+		if spell.ProcMask.Matches(ProcMaskWhiteHit) {
+			swingSpeed := 0.0
+			baseMultiplier *= 0.4
+			if spell.IsMH() {
+				mh := spell.Unit.AutoAttacks.MH()
+				if mh != nil {
+					swingSpeed = mh.SwingSpeed
+				}
+			} else {
+				baseMultiplier /= 2
+				oh := spell.Unit.AutoAttacks.OH()
+				if oh != nil {
+					swingSpeed = oh.SwingSpeed
+				}
+			}
+			speedMultiplier = swingSpeed / 2.6
```

Comment:
Do you also have to handle the ranged auto case where you use the ranged swing timer for normalization?